### PR TITLE
Loki: Update Query Variable docs

### PR DESCRIPTION
**Pull request by matyax**
**What this PR does / why we need it**:

From [here](https://github.com/grafana/grafana/pull/54625#pullrequestreview-1100695191).

This PR updates the docs to reflect recent changes in the datasource.

**Which issue(s) this PR fixes**:

Part of https://github.com/grafana/grafana/issues/52380

**Special notes for your reviewer**:

Type Label names:

imagen

Type Label values:

imagen

### CODE REVIEW

File: docs/sources/datasources/loki.md [hunk 1]
```diff
@@ -213,13 +213,14 @@ Check out the [Templating]({{< relref "../variables/" >}}) documentation for an
...
-| ------------------------------------------ | -------------------------------------------------------------------------------------- |
-| `label_names()`                            | Returns a list of label names.                                                         |
-| `label_values(label)`                      | Returns a list of label values for the `label`.                                        |
-| `label_values(log stream selector, label)` | Returns a list of label values for the `label` in the specified `log stream selector`. |
+provides a form to select the type of values that can be expected to be used for a given variable. This form has the following
+options to choose from:
+
+| Query Type   | Label       | Stream Selector       | Description                                                                            |
+| ------------ | ----------- | --------------------- | -------------------------------------------------------------------------------------- |
+| Label names  | No required | Not required          | Returns a list of label names.                                                         |
```
Comment (ivanahuckova) [inline docs/sources/datasources/loki.md:221]:
```suggestion
| Label names  | Not required | Not required          | Returns a list of label names.                                                         |
```

**Review by ivanahuckova — APPROVED** [review]
LGTM! Looks good, but I'd wait for a feedback from @grafana/docs-squad.

File: docs/sources/datasources/loki.md (hunk 1 above)
Comment (matyax) [reply docs/sources/datasources/loki.md:221]: Thanks!

File: docs/sources/datasources/loki.md [hunk 2]
```diff
@@ -213,13 +213,14 @@ Check out the [Templating]({{< relref "../variables/" >}}) documentation for an
...
 Variable of the type _Query_ allows you to query Loki for a list labels or label values. The Loki data source plugin
-provides the following functions you can use in the `Query` input field.
-
-| Name                                       | Description                                                                            |
-| ------------------------------------------ | -------------------------------------------------------------------------------------- |
-| `label_names()`                            | Returns a list of label names.                                                         |
-| `label_values(label)`                      | Returns a list of label values for the `label`.                                        |
-| `label_values(log stream selector, label)` | Returns a list of label values for the `label` in the specified `log stream selector`. |
+provides a form to select the type of values that can be expected to be used for a given variable. This form has the following
+options to choose from:
```
Comment (knylander-grafana) [inline docs/sources/datasources/loki.md:217]: The second sentence ("The Loki data source...") was a little hard to parse. I tried to clarify the wording on this sentence. Is this still correct? 
```suggestion
provides a form to select the type of values that can be expected for a given variable.
The form has these options:
```

**Review by knylander-grafana — APPROVED** [review]
Approving with one minor suggestion

File: docs/sources/datasources/loki.md (hunk 2 above)
Comment (matyax) [reply docs/sources/datasources/loki.md]: Updated. Thanks!
